datalad_osf/remote.py

```diff
@@ -181,11 +181,58 @@ def files(self):
             # per-request latency is substantial, presumably it is overall
             # faster to get all at once
             self._files = {
+                # TODO use f.path to make it as useful for exports as well
                 f.name: f
                 for f in self.storage.files
             }
         return self._files
 
+    def transferexport_store(self, key, local_file, remote_file):
+        # store the file located at `local_file` to `remote_file` on the remote
+        # raise RemoteError if the file couldn't be stored
+        with open(local_file, 'rb') as fp:
+            # there is no need to pre-created subdirectories, all done
+            # automatically inside
+            self.storage.create_file(remote_file, fp, force=True, update=True)
+
+    def transferexport_retrieve(self, key, local_file, remote_file):
+        # get the file located at `remote_file` from the remote and store it to
+        # `local_file`
+        # raise RemoteError if the file couldn't be retrieved
+        self.annex.info(local_file)
+        self.annex.info(remote_file)
+        # find the right file object
+        fobj = [f for f in self.files.values() if f.path == posixpath.sep + remote_file]
+        # git-annex seems to provide the receiving directory
+        with open(local_file, 'wb') as fp:
+            fobj[0].write_to(fp)
+
+    def checkpresentexport(self, key, remote_file):
+        # return True if the file `remote_file` is present in the remote
+        # return False if not
+        # raise RemoteError if the presence of the file couldn't be determined,
+        # eg. in case of connection error
+        self.annex.info(key)
+        self.annex.info(remote_file)
+        return posixpath.sep + remote_file in set(f.path for f in self.files)
```

Comment:
This breaks for me, probably because the representation of self.files changed to a dictionary.

Reply:
Right. Should be `self.files.values()`

Reply:
FTR: All of this code is gone now.